src/shipit_code_coverage/shipit_code_coverage/utils.py

```diff
@@ -12,3 +13,19 @@ def wait_until(operation, timeout=30):
         elapsed += 1
 
     return None
+
+
+class ThreadPoolExecutorResult(ThreadPoolExecutor):
```

Comment:
I introduced this wrapper class for ThreadPoolExecutor to call `result` on all futures created with `submit`, this way we don't miss errors in workers.